tests/rptest/tests/compatibility/kafka_streams_test.py

```diff
@@ -550,3 +551,197 @@ def test_kafka_streams_map_function(self):
         consumer.stop()
         self._producer.stop()
         example.stop()
+
+
+class KafkaStreamsInteractiveQueries(RedpandaTest):
+    """
+    Test KafkaStreams InteractiveQueries app which locates and
+    queries kafka streams state stores. The state stores used in this app are
+    a KeyValue store and a Windowed Store.
+
+    Outside of KStreams, the app does simple prod-cons so the used RP features and systems are:
+    - topic creation
+    - produce/consume on topic
+    - src/v/kafka
+    - src/v/raft
+    - src/v/storage
+    """
+
+    topics = (TopicSpec(name="TextLinesTopic"), )
+
+    def __init__(self, test_context):
+        super(KafkaStreamsInteractiveQueries,
+              self).__init__(test_context=test_context)
+        self._ctx = test_context
+        self._ports = [7070, 7071]
+        self._store_names = ["windowed-word-count", "word-count"]
+
+    def list_all_running_instances(self, res):
+        # Example response looks like
+        # [{'host': 'docker_n_4', 'port': 7070, 'storeNames': ['windowed-word-count']},
+        # {'host': 'docker_n_5', 'port': 7071, 'storeNames': ['word-count']}]
+
+        if len(res) != 2:
+            return False
+
+        if len(res[0]["storeNames"]) != 1 or len(res[1]["storeNames"]) != 1:
+            return False
+
+        valid_host = "docker_n_" in res[0]["host"] and "docker_n_" in res[1][
+            "host"]
+
+        valid_port = res[0]["port"] in self._ports and res[1][
+            "port"] in self._ports
+
+        valid_store_name = res[0]["storeNames"][0] in self._store_names and res[
+            1]["storeNames"][0] in self._store_names
+
+        return valid_host and valid_port and valid_store_name
```

Comment:
It's simpler if you use this pattern:
 - lesser change to forget to check e.g. `res[1]`
 - lesser conjunctions => easier to understand what's going on
 - precise checking of host instead of the approximate 

```
for item in res:
    if not re.match("^docker_n_\d+$", item["host"]):
        return False
    if item["port"] not in self._ports:
        return False
    if len(item["storeNames"])!=1:
        return False
    if item["storeNames"][0] not in self._store_names:
        return False
return True
```